common/util.go

```diff
@@ -387,3 +395,15 @@ func JoinURL(url, path string) string {
 	}
 	return url + path
 }
+
+// Read at most n bytes from an io.Reader
+// error out if the io.Reader had bytes remaining
+func ReadAtMost(r io.Reader, n int) ([]byte, error) {
+	limitedReader := io.LimitReader(r, int64(n)+1)
```

Comment:
Should the limit be n as opposed to n + 1?

Reply:
If we keep it as `n`, `len(b)` in the subsequent check will always be `<= n`, and we won't be able to find out if the the underlying `io.Reader` had more bytes than we want to read (i.e. `n`)

I'll add a comment to make this clearer.